.github/workflows/application-signals-java-e2e-k8s-test.yml

```diff
@@ -120,6 +131,15 @@ jobs:
           curl -S -s "http://${{ env.MAIN_SERVICE_ENDPOINT }}:30100/remote-service?ip=${{ env.REMOTE_SERVICE_IP }}&testingId=${{ env.TESTING_ID }}"; echo
           curl -S -s "http://${{ env.MAIN_SERVICE_ENDPOINT }}:30100/client-call"; echo
 
+      - name: Initiate Gradlew Daemon
+        if: steps.initiate-gradlew == 'failure'
+        uses: ./.github/workflows/actions/execute_and_retry
+        with:
+          command: "./gradlew"
```

Comment:
Why does it matter if the first try failed? Does this try to overwrite the previous attempt?

Check for options that would allow `./gradlew` to just stop if the job is already done.

Also, should we `continue-on-error: true` here as well since we try again when validating too?

Reply:
1. If the first try failed then we want to run it again to make sure `./gradlew` creates a daemon successfully. That's the point of splitting it to the beginning of the run and right before the validation step
2. That's basically what the `if` statement is doing. If it succeeded don't run ./gradlew again, if failed run it again.
3. Makes sense. Will add